### PR TITLE
Feature/US-890 OrderNumber

### PR DESCRIPTION
**Pull request by astoneDTE**

### CODE REVIEW

File: app/components/OrderNumber/formatOrderNumber.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default function formatOrderNumber(value: string) {
+
+    const formattedOrderNumber = `${value.substr(0,4)} ${value.substr(4,4)} ${value.substr(8,3)} ${value.substr(11,1)}`;
```
Comment (JeremySparkman) [inline app/components/OrderNumber/formatOrderNumber.ts:3]: since we're only declaring formattedOrderNumber to return it we could instead just return our formatted string

File: app/components/OrderNumber/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const OrderNumber: React.FC<IOrderNumberProps> = (props: IOrderNumberProps) => {
+  const { value } = props;
+
+  if (value.length !== 12) {
+    throw new Error('order numbers must be exactly 12 digits.');
+  }
+
+  const formattedOrderNumber = formatOrderNumber(value);
```
Comment (JeremySparkman) [inline app/components/OrderNumber/index.tsx:17]: Also same here, we could just render formatOrderNumber(value). Not a big deal but saves space.

**Comment by JeremySparkman** [comment]
Looks good, just added some smalls suggestions

File: app/components/OrderNumber/formatOrderNumber.ts (hunk 1 above)
Comment (jmiller-nexient) [reply app/components/OrderNumber/formatOrderNumber.ts:3]: Yea, I agree. Not much value in having this as a separate file in this case.

File: app/components/OrderNumber/index.less [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@import '../../styles/themes/index.less';
+
+.dte-wismo-order-number {
+  color: @dte-gray-500;
+  font-family: @dte-wismo-font-family;
+  font-size: @dte-wismo-font-size-1;
+}
```
Comment (jmiller-nexient) [inline app/components/OrderNumber/index.less:7]: Remove this.  We don't want to apply styles to the raw components as that decreases their usefulness.  The only thing you are specifying here that isn't already defined in the theme for all text is the color and we want the parent to define that.  Doing so ensures a high-level of reusability for the component.

File: app/components/OrderNumber/tests/formatOrderNumber.test.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import formatOrderNumber from '../formatOrderNumber';
+
+describe('function formatOrderNumber(value)', () => {
+    it('should format a string correctly', () => {
+        const actual: string = formatOrderNumber('123456789012');
+        const expected = '1234 5678 901 2';
+        expect(actual).toEqual(expected);
+    });
+});
```
Comment (jmiller-nexient) [inline app/components/OrderNumber/tests/formatOrderNumber.test.ts:9]: Add tests for negative cases as well.  For instance, less than or more than 12 digits and strings that contain non-numeric characters.

File: app/components/OrderNumber/index.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import './index.less';
+
+interface IOrderNumberProps {
+  value: string;
+}
+
+const OrderNumber: React.FC<IOrderNumberProps> = (props: IOrderNumberProps) => {
+  const { value } = props;
+
+  if (value.length !== 12) {
```
Comment (jmiller-nexient) [inline app/components/OrderNumber/index.tsx:13]: A better test would use a regex to ensure there are 12 numeric characters only.  This checks the length but doesn't ensure they are all numeric.

File: app/components/OrderNumber/tests/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React, { ReactElement } from 'react';
+import renderer from 'react-test-renderer';
+
+import OrderNumber from '..';
+import {shallow} from 'enzyme';
+
+describe('<OrderNumber />', () => {
+  it('should render correctly', () => {
```
Comment (jmiller-nexient) [inline app/components/OrderNumber/tests/index.test.tsx:8]: Always provide a "when" clause so it is clear what scenario is being tested.
